compiler/rustc_resolve/src/late.rs

```diff
@@ -3315,15 +3315,19 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
         if let Some(qself) = &delegation.qself {
             self.visit_ty(&qself.ty);
         }
+        let last_segment = delegation.path.segments.last().unwrap();
+        let traits = self.traits_in_scope(last_segment.ident, ValueNS);
+        // Saving traits for a `MethodCall` that has not yet been generated.
+        self.r.delegation_trait_map.insert(delegation.id, traits);
```

Comment:
I don't think a separate map is necessary, this can be inserted into the existing `trait_map`.

Reply:
It can't. HIR trait map is filled during `NodeId` lowering from AST trait map. For a delegation item method call is generated after late resolution and `NodeId` in AST trait map is not equal to method call  `NodeId`.

Reply:
That doesn't contradict my suggestion.
Two entries (for both the delegation item and the method call) can be added to the same `trait_map`.